core/chains/evm/logpoller/log_poller.go

```diff
@@ -635,8 +639,9 @@ func (lp *logPoller) backfill(ctx context.Context, start, end int64) error {
 		}
 
 		lp.lggr.Debugw("Backfill found logs", "from", from, "to", to, "logs", len(gethLogs), "blocks", blocks)
-		err = lp.orm.q.WithOpts(pg.WithParentCtx(ctx)).Transaction(func(tx pg.Queryer) error {
-			return lp.orm.InsertLogs(convertLogs(gethLogs, blocks, lp.lggr, lp.ec.ChainID()), pg.WithQueryer(tx))
+		err = lp.saveLogsInTx(ctx, func(tx pg.Queryer) ([]Log, error) {
+			logs := convertLogs(gethLogs, blocks, lp.lggr, lp.ec.ChainID())
+			return logs, lp.orm.InsertLogs(logs, pg.WithQueryer(tx))
```

Comment:
Why is it necessary to pass a function here? Why not pass the logs?

Reply:
The reason I used a fn is because `PollAndSaveLogs` inserts a block into the DB in the same tx as inserting the logs, but here, in `backfill`, only logs are inserted.

Alternatively, I could pass the logs as an additional argument into the `saveLogsInTx` rather than have the fn return the logs. But it seemed to make a bit more sense to me that the fn would return the logs that it saved to the DB instead of passing both the logs and a function that saves the logs. Happy to change if you think otherwise.